exec/selected/selected.go

```diff
@@ -55,6 +55,30 @@ type SchemaField struct {
 	FixedResult reflect.Value
 }
 
+func (sf *SchemaField) ToSelectedField() *types.SelectedField {
+	if sf == nil {
+		return nil
+	}
+	return &types.SelectedField{
+		Name:       sf.Name,
+		Alias:      sf.Alias,
+		Fields:     selsToSelectedFields(sf.Sels),
+		Args:       sf.Args,
+		Directives: sf.Directives,
+	}
+}
+
+func selsToSelectedFields(sels []Selection) (fs []*types.SelectedField) {
+	for _, f := range sels {
+		v, ok := f.(*SchemaField)
```

Comment:
pls confirm this covers all of our use cases

previously you had a switch here

does this work with fragments?

Reply:
Yes it does work with fragments (double checked it) but I've also added `TypeAssertion` handling for union types